axlearn/experiments/aot_test.py

```diff
@@ -0,0 +1,54 @@
+# Copyright © 2023 Apple Inc.
+
+"""AoT (ahead-of-time) compilation config tests.
+
+pip install 'jax[tpu]==0.4.21' -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
+
+export TPU_SKIP_MDS_QUERY=1
+python axlearn/experiments/aot_test.py
+
+Reference:
+https://docs.google.com/document/d/1Y5IdmvAZA7UtMHAWkRh8k2PscVoG5FvMH9-E6hygsyY/
+"""
+import pickle
+from typing import Optional
+
+from absl.testing import absltest
+from jax.experimental.serialize_executable import serialize
+
+from axlearn.common import test_utils
+from axlearn.common.aot_compilation import compile_trainer_programs
+from axlearn.common.trainer import SpmdTrainer
+from axlearn.experiments.text.gpt import c4_trainer
+
+
+class AoTCompilationTest(test_utils.TrainerConfigTestCase):
+    """Tests ahead-of-time (AoT) compilation."""
+
+    def _jax_backend(self) -> str:
+        return "cpu"
```

Comment:
nit: consider removing if it's going to use the default anyway?

Reply:
Done.